exchange/exchange.go

```diff
@@ -698,15 +691,21 @@ func (e *exchange) getAllBids(
 			reqInfo.PbsEntryPoint = bidderRequest.BidderLabels.RType
 			reqInfo.GlobalPrivacyControlHeader = globalPrivacyControlHeader
 			reqInfo.BidderRequestStartTime = start
-
-			bidReqOptions := bidRequestOptions{
-				accountDebugAllowed: accountDebugAllowed,
-				headerDebugAllowed:  headerDebugAllowed,
-				addCallSignHeader:   isAdsCertEnabled(experiment, e.bidderInfo[string(bidderRequest.BidderName)]),
-				bidAdjustments:      bidAdjustments,
+			respInfo := &bidderResponseInfo{}
+			requestBidInfo := extraRequestBidInfo{
+				request: bidderRequestInfo{
+					accountDebugAllowed: accountDebugAllowed,
+					headerDebugAllowed:  headerDebugAllowed,
+					addCallSignHeader:   isAdsCertEnabled(experiment, e.bidderInfo[string(bidderRequest.BidderName)]),
+					bidAdjustments:      bidAdjustments,
+				},
+				response: respInfo,
 			}
-			seatBids, err := e.adapterMap[bidderRequest.BidderCoreName].requestBid(ctx, bidderRequest, conversions, &reqInfo, e.adsCertSigner, bidReqOptions, alternateBidderCodes, hookExecutor, bidAdjustmentRules)
 
+			seatBids, err := e.adapterMap[bidderRequest.BidderCoreName].requestBid(ctx, bidderRequest, conversions, &reqInfo, e.adsCertSigner, requestBidInfo, alternateBidderCodes, hookExecutor, bidAdjustmentRules)
+			if err == nil {
```

Comment:
`requestBid()` returns an error array. There's a possibility that it contains warnings and/or errors and valid bids at the same time:
```
   |           seatBids                 |       err          ||
---|------------------------------------|--------------------||------------------------
 1 | [ {Bids:[{...}]}, {Bids:[{...}]} ] | [ ]                || Records metrics
 2 | [ {Bids:[{...}]}, {Bids:[{...}]} ] | [ error ]          || Won't record metrics
 3 | [ {Bids:[{...}]}, {Bids:[{...}]} ] | [ warning ]        || Won't record metrics
 4 | [ {Bids:[{...}]}, {Bids:[{...}]} ] | [ error, warning ] || Won't record metrics
```
Do we want this behavior? Or should we check for elements on `seatBids` instead?
```
708             seatBids, err := e.adapterMap[bidderRequest.BidderCoreName].requestBid(ctx, bidderRequest, conversions, &reqInfo, e.adsCertSigner, requestBidInfo, alternateBidderCodes, hookExecutor, bidAdjustmentRules)
709 -           if err == nil {
    +           if len(seatBids) > 0 {
710                 brw.bidderResponseStartTime = respInfo.bidderResponseStartTime
711             }
exchange/exchange.go
```
Or maybe check for fatal errors:
```
708             seatBids, err := e.adapterMap[bidderRequest.BidderCoreName].requestBid(ctx, bidderRequest, conversions, &reqInfo, e.adsCertSigner, requestBidInfo, alternateBidderCodes, hookExecutor, bidAdjustmentRules)
709 -           if err == nil {
    +           if !errortypes.ContainsFatalError(err) {
710                 brw.bidderResponseStartTime = respInfo.bidderResponseStartTime
711             }
exchange/exchange.go
```
Maybe we should also rename `err` to `errs` or `errL` just to make it extra clear that it is an error array. Thoughts?

Reply:
Interesting point. There are can be valid bids and/or errors and/or warnings. And any combination of them is a response. Maybe this if is not needed here, just 
```go
seatBids, err := e.adapterMap[bidderRequest.BidderCoreName].requestBid(ctx, bidderRequest, conversions, &reqInfo, e.adsCertSigner, requestBidInfo, alternateBidderCodes, hookExecutor, bidAdjustmentRules)	
brw.bidderResponseStartTime = respInfo.bidderResponseStartTime
```

Reply:
`bidderResponseStartTime` holds the time of when processing begins for last successful response received bidder server.
 
https://github.com/prebid/prebid-server/blob/e3c34fdd92d18f0568077bfdb2bc5cf26b9e034f/exchange/bidder.go#L238-L239

So instead of checking `seatBids` or `err` how about checking if `bidderResponseStartTime` returned from `requestBid` is not Zero

https://github.com/prebid/prebid-server/blob/e3c34fdd92d18f0568077bfdb2bc5cf26b9e034f/exchange/exchange.go#L697-L700